File: Course5_BackEnd(kotlin_spring)/ch02_springNkotlin/java-kotlin/src/main/java/org/example/ex09/Exam09.java
package org.example.ex09;

public class Exam09 {
    public Exam09(StoreUser storeUser) {
        // service logic
        if (("MASTER").equals(storeUser.getRole())) {
            //master
        } else if (("ADMIN").equals(storeUser.getRole())) {
            //admin
        } else if (("USER").equals(storeUser.getRole())) {
            // user
        } else {
            // default
        }

        var role = "";
        switch (storeUser.getRole()) {
            case "MASTER":
            case "ADMIN":
                role = "MASTER";
                break;
            case "USER":
                break;
            default:
                // default
        }

        try {

        } catch (Exception e) {
            var result = "";
            if (e instanceof NullPointerException) {

            } else if (e instanceof NumberFormatException) {
            }
        }
    }

    public static void main(String[] args) {
        var storeUser = new StoreUser();
        new Exam09(storeUser);
    }
}

class StoreUser {
    private String name;
    private String role;

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getRole() {
        return role;
    }

    public void setRole(String role) {
        this.role = role;
    }
}

File: Course5_BackEnd(kotlin_spring)/ch01_kotlin/kotlin-example/src/main/kotlin/ex09/Exam09.kt
package ex09

fun main() {

    var result = when ("") {
        "" -> {
            ""
        }
        "MASTER", "ADMIN" -> {
            "master"
        }

        "USER" -> {
            "user"
        }

        else -> {
            "default"
        }
    }
    println(result)

    var any: Any = ""
    var exception: RuntimeException

    when (any) {
        is NullPointerException -> {
            // type casting, instanceof
        }
        is RuntimeException -> {

        }
    }

    var number = 13

    when (val n = number % 2) {
        0 -> {
            println(n)
        }
        else -> {
            println(n)
        }
    }

    var r = when{
        number % 2 == 0 ->{
            100
        }
        else -> {
            200
        }
    }
    println(r)
}